workflows/ncn/hooks/before-all/move-critical-singleton-pods.yaml

```diff
@@ -52,24 +52,20 @@ spec:
 
     echo
 
-    containerNames=( "nexus" "ipxe" "kea" "cray-cfs-api-db" )
-    for containerName in ${containerNames[@]};do
-      # get pod name by container name
-      podName=$(kubectl get pods -A -o=jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.status.phase}{"\t"}{range .spec.containers[*]}{.name}{"\t"}{end}{"\n"}{end}' | grep -E "^[^[:space:]]+[[:space:]]+Running[[:space:]]+.*${containerName}.*$" | cut -f1)
-      if [[ -z $podName ]];then
-        echo "Couldn't not find pod: $pod"
-        echo "something is wrong"
-        exit 1
-      fi                 
-      
-      currentNode=$(kubectl get po -A -o wide | grep $podName | awk '{print $8}')
-      if [[ "$nodeMoveTo" != "$currentNode" ]]; then
-        echo "Move Pod: $podName to Node: $nodeMoveTo"
-        /opt/cray/platform-utils/move_pod.sh $podName $nodeMoveTo > /dev/null
-      fi
+    criticalPodLables=( "app=nexus" "app.kubernetes.io/name=cray-ipxe" "app.kubernetes.io/name=cray-dhcp-kea" "app.kubernetes.io/name=cray-cfs-api-db" )
+    for label in ${criticalPodLables[@]};do
+      pods=($(kubectl get pods -A -l ${label} -o custom-columns=:.metadata.name))
+      # we may have multiple pods matched with the lable
```

Comment:
The name of this script is "move-critical-singleton-pods".  Should we log or fail if we find multiple pods in this case since that is unexpected?

Reply:
i think i will rename the file? it will be future proofing when we want to add specific label  like `critical:true`.